bundles/framework/divmanazer/component/Chart.js

```diff
@@ -134,29 +195,95 @@ Oskari.clazz.define('Oskari.userinterface.component.Chart', function() {
             this.containerWidth = options.width;
         }
     },
+    dataHasNegativeValues: function () {
+        var dataset = this.getDatasetMinMax();
+        return dataset.min <= 0;
+    },
     /**
      *
-     * @method callGroups
+     * @method svgAppendElements
      */
-    callGroups: function () {
-        //groups
-        var gx = this.svg.append( "g" )
-            .attr("class", "x axis" )
-            .attr("transform", "translate(0 " + this.dimensions.xAxisOffset + ")")
-            .call( this.xAxis );
+    svgAppendElements: function () {
+        var me = this;
+
+        if( this.dataHasNegativeValues() ) {
+            var negatives = [];
+            this.data.forEach( function (d) {
+                if(d.value < 0) {
+                    negatives.push( d.name );
+                }
+            });
+        }
+        // append x-tick lines to chart
+        var xtickAxis = this.svg.append("g")
+            .attr("class", "x axis")
+            .attr("transform", "translate( 0, 0)")
+            .call(this.xAxis);
+
+        xtickAxis.selectAll("x axis, tick, text").remove();
+        xtickAxis.select('.domain').remove();
+
+
+        var gy = this.svg.append("g")
+            .attr("class", "y axis")
+            .attr("transform", "translate(" + this.x(0) + ", 0)")
+            .call(this.yAxis);
 
-        gx.select('.domain').remove();
-        gx.selectAll('line, path')
-            .attr('stroke', '#aaa')
-            .attr('shape-rendering', 'crispEdges');
-            
-        var gy = this.svg.append( "g" )
-            .attr( "class", "y axis" )
-            .call( this.yAxis );
         gy.selectAll('line, path')
+            .attr('x2', function ( d ) {
+                if ( d === null ) {
+                    return;
+                }
+                if ( negatives !== undefined ) {
+                    if ( negatives.indexOf( d ) !== -1 && negatives !== undefined ) {
+                        return 5;
+                    } else {
+                        return -5;
+                    }
```

Comment:
the negatives array is there because "this.y" domain only contains the names of the datapoints, not the actual values.
https://github.com/nxholm/oskari-frontend/blob/statsgrid/diagram-grid-changes/bundles/framework/divmanazer/component/Chart.js#L69

Reply:
Yes, I see. Maybe in this complex y-axis rendering  custom code for drawing the y-axis ticks & labels would be better? I mean creating a new SVG group and binding the actual data in `this.data` to some new elements in the same way the bars are rendered. This way we have access to the whole datapoint; name and value.

Another solution would be to change Chart.js to use the whole data array as the domain (instead of mapping out just the names). For this to work, we'd need to implement toString() on the datapoint objects, though:
https://github.com/d3/d3-scale#ordinal_domain